### PR TITLE
timers: Adapt and enable Windows timer support on Xbox

### LINKED ISSUE #17: Timer dummy backend returns constant values
We currently use "dummy" for the SDL timers backend. However, that returns a constant 0 for `SDL_GetTicks()` (It doesn't handle intrinsics for CPU based timing and doesn't even use a static variable to increment the ticks).

This breaks assumptions in applications. Ideally this wouldn't be a problem in SDL, but because it is, we should switch backends as soon as possible.

We should try to use the "windows" timer backend or bring our own "xbox" backend.

### PR DESCRIPTION
**Pull request by JayFoxRox**
Closes #17 

**This code has not been tested yet.**

### CODE REVIEW

File: src/timer/windows/SDL_systimer.c [hunk 1]
```diff
@@ -23,7 +23,11 @@
 #ifdef SDL_TIMER_WINDOWS
 
 #include "../../core/windows/SDL_windows.h"
+#if !defined(__XBOX__)
 #include <mmsystem.h>
+#else
+typedef unsigned int UINT;
```
Comment (JayFoxRox) [inline src/timer/windows/SDL_systimer.c]: This should be in WinDef.h according to https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types#uint

Ideally we'd do this first, so we don't have the type declared twice in the future.

Comment (JayFoxRox) [reply src/timer/windows/SDL_systimer.c]: **Waiting for https://github.com/XboxDev/nxdk/pull/287**

Comment (JayFoxRox) [reply src/timer/windows/SDL_systimer.c]: That PR got merged, so I removed the `typedef` from this PR.

**Comment by dracc** [comment]
Works for me in xqemu. Only tested `SDL_Delay();`, but I guess that's as good as any test for this change.

**Comment by thrimbor** [comment]
Tested it with `SDL_Delay()` and the timer callbacks, seems to work fine. Merging.

**Pull request merged**